provider/schema/execution.go

```diff
@@ -24,6 +25,23 @@ import (
 // faster than the <1s it won't be deleted by remove stale.
 const executionJitter = -1 * time.Minute
 
+//go:generate mockgen -package=mock -destination=./mock/mock_database.go . Database
+type Database interface {
+	QueryExecer
+
+	Insert(ctx context.Context, t *Table, instance Resources) error
+	Delete(ctx context.Context, t *Table, kvFilters []interface{}) error
+	RemoveStaleData(ctx context.Context, t *Table, executionStart time.Time, kvFilters []interface{}) error
+	CopyFrom(ctx context.Context, resources Resources, shouldCascade bool, CascadeDeleteFilters map[string]interface{}) error
+	Close()
+	Dialect() Dialect
+}
+
+type QueryExecer interface {
+	Exec(ctx context.Context, query string, args ...interface{}) error
+	Query(ctx context.Context, query string, args ...interface{}) (pgx.Rows, error)
+}
```

Comment:
I think the naming is a little off now also would move these to database package as these more sense there. I feel there is no way other than moving execution and schema into separate packages. Then Database interfaces can sit in schema but be named differently perhaps. (ResourceInserter?)

Reply:
It's not just for insertion, policies and drift also uses QueryExecer in cq-core as now it's the only way to access the DB.

Reply:
I agree, I think my comments confused a little bit.  Would move QueryExecer -> database package (as like you said core packages use it and its a database thing). The Database interface is only used by execution as its an execution thing, and then I would rename it perhaps.

Reply:
`Database` is now called `Storage`, and `QueryExecer` now includes `pgxscan.Querier` instead of our `Query` signature... (Exec is still there as-is)